week-3/Homework/mandatory/5-journey-planner.js

```diff
@@ -1,24 +1,172 @@
+
+//   This technic is also referred as problem decomposition. It helps you to reduce scope of the problem 
+//   by only focusing on a small chunk of the whole problem at a time.)
+// */
+
+/*
+  Implement the function getTransportModes that
+   - Accepts an array containing the location and available transport modes
+     e.g: ["Tower Bridge", "tube", "river boat"]
+   - Returns an array including the available transport modes to the given location
+     e.g: ["tube", "river boat"]
+  
+  Hint: Use the corresponding array method to split the array.*/
+
+
+function getTransportModes(meansOfTransport) {
+  var newArr=meansOfTransport.splice(1,2);
+  return newArr;
+  }
+
+  
+ 
+ console.log(getTransportModes(["Tower Bridge", "tube", "river boat"]));
+ 
+
+/*
+  Implement the function isAccessibleByTransportMode that
+   - Accepts two parameters:
+     1) First parameter is an array of transport modes
+        e.g: ["tube", "river boat"]
+     2) Second parameter is a string containing a transport mode
+        e.g: "river boat"
+     
+   - Returns 
+     * True if the location in the first parameter is accessible by the transport mode given in second parameter
+     * Otherwise, returns false
+
+  Hint: Use the corresponding array method to decide if an element is member of an array.
+*/
+function isAccessibleByTransportMode(newArr,transport) { 
+  if(newArr.includes(transport)){
+    return true;
+  }
+  else{
+    return false;
+  }
+}
+
+/*
+  Implement the function getLocationName that
+   - Accepts a location and available transports in an array
+     e.g:["Tower Bridge", "tube", "river boat"]
+   - Returns the name of the location
+     e.g: "Tower Bridge"
 */
+function getLocationName(newArr) { 
+  var nameOfLocation=newArr.slice(0,1).shift();
+  return nameOfLocation;
+}
+console.log(getLocationName["Tower Bridge", "tube", "river boat"]);
+
+/*
+ We arrived at the final method. it won't take long if you use the previously implemented functions wisely.
 
-function journeyPlanner() {
+ Finish up the implemention of the function journeyPlanner that
+  - Accepts two parameters:
+    1) An array with a list of locations' and their transports
+       e.g:
+       [
+         ["Angel", "tube", "bus"],
+         ["London Bridge", "tube", "river boat"]
+       ]
+    2) A string containing a transport mode
+       e.g: "bus"
 
+  - Returns an array of where I can go if I only want to use a specific mode of transport.
+    NOTE: only the location names should be returned, not the name of transports.
+
+  Hint:
+   - Use the function you implemented above.
+   - Use array method to remove locations that is not accessible by the given transportMode.
+   - Use array method to manipulate its elements.
+   
+  Advanced challange: try to use arrow function when invoking an array method.
+*/
+function journeyPlanner(locations, transportMode) {
+  // Implement the function body
+  return locations.filter(b => b===transportMode);
```

Comment:
So you need to filter here using the function you made earlier - think of how you can use `isAccessibleByTransportMode` in order to filter these nested arrays. I can talk through this solution if you'd like - it is a little tricky.